### ISSUE TITLE
fps

### ISSUE THREAD

**Issue by doctorten10** [ChainSwordCS/CHokiMod#9]
installed all the cia's recently, original run, on an old 3ds, i was able to connect and stream. The issue is, I would only get one or two fps. That was last night. Today it doesnt want to connect. i run chiruno on my 3ds and the rgb led is lit up bright blue, but it doesnt want to connect to my pc
![Screenshot (1)](https://github.com/ChainSwordCS/ChirunoMod/assets/168360265/15d37104-cff2-4f7f-a7db-abf884fb2613)
![Screenshot (2)](https://github.com/ChainSwordCS/ChirunoMod/assets/168360265/0312eda8-f4d0-45b5-b49a-94e01a26d268)
![Screenshot (3)](https://github.com/ChainSwordCS/ChirunoMod/assets/168360265/38b2c3ab-e4e1-4f92-84e1-b9adbc61ad7b)
![Screenshot (4)](https://github.com/ChainSwordCS/ChirunoMod/assets/168360265/bdf19212-aece-464c-ac8e-f89470dd1a22)

**Comment by doctorten10** [comment]
Hello?

**Comment by Eiim** [comment]
ChainSwordCS doesn't always monitor GitHub actively. I'll send him a message.

**Comment by doctorten10** [comment]
> ChainSwordCS doesn't always monitor GitHub actively. I'll send him a message.

any response?

**Comment by doctorten10** [comment]
Hello?

**Comment by ChainSwordCS** [comment]
Hello! Apologies for not getting back to you sooner.

You mentioned two issues. So firstly, the low FPS on Old 3DS. Unfortunately, that's about how well it's known and expected to work. The Old 3DS doesn't have much memory or CPU power to spare. Though my code has room for improvement in performance, and I'll check to see whether performance is significantly worse than HzMod. But Old 3DS will probably only ever hit 4-8 or maybe 10 FPS at best.

Secondly, the "Connection Timed Out" issue with Chokistream. Did you try checking to make sure you input the correct local IP address that matches the 3DS?

One way of finding the local IP address of the 3DS is to open the Rosalina menu (L+Down+Select), select "Miscellaneous options...", then "Start InputRedirection", then press B, and the IP address should appear in the top right corner of the bottom screen. Assuming you're connected to wifi.

**Comment by doctorten10** [comment]
First off, Thank you for getting back to me at all... I had actually forgotten about this thread completely, as I had to deal with other things irl, as well as dealing with autism/adhd brain. When I logged in just now and checked my notifications, I was expecting a response from a few months back... not 4 days lmao.
Secondly, thank you for the advice about Rosalina menu, I will go ahead and check things out now.

**Comment by Professor-Marill** [comment]
Would it be possible to convert the O3ds Chiruno mod from tcp to udp. The new 3ds's still have a harder time while running it than Boot NTR. The main difference that I know of is that HzMod utilized tcp and BootNTR used udp. So would it be possible to get a streaming client that works on the O3DS that uses udp instead? And wouldnt that at least help some?

**Comment by Eiim** [comment]
Yes, that is a known avenue for improvement. It's not particularly easy to rewrite HzMod/ChirunoMod to use UDP, but he'd like to do it sometime. It also requires simultaneous rewrites in Chokistream to handle UDP there, but that should be less of an issue since we already have a process for it with NTR.

It's also worth noting that TCP/UDP is not necessarily the "main difference" between NTR and HzMod/ChirunoMod. They interact with the 3DS in very different ways, which is why NTR doesn't support O3DS, but is also probably one major reason it's faster. They also have different features when it comes to video encoding, which affects performance.

**Comment by ChainSwordCS** [comment]
What Eiim said. Though I'd like to add:

TCP/UDP _is_ kinda the main difference. That's why NTR 3.6.1 on n3DS hits up to 30-40 FPS in situations where HzMod and ChirunoMod hit up to 20-30 FPS. Furthermore, NTR-HR takes advantage of multiple cores to generate JPEGs, so it can actually hit up to 55-60 FPS in those situations if the filesize of each JPEG is small enough, whereas HzMod and ChirunoMod are bottlenecked by network communications (assuming similarly small JPEGs).

It's true that there are other major differences in how ChirunoMod and NTR at the lower level, though. Which is why, for example, they have different compatibility with certain games and homebrew. But anyways...

I haven't added UDP as a feature yet because I haven't figured out how to make it work, but it's on the to-do list. But I still need to rewrite some things before that project makes sense to tackle.
